pkg/operator/bootstrapteardown/waitforceo.go

```diff
@@ -73,3 +83,114 @@ func done(etcd *operatorv1.Etcd) (bool, error) {
 	klog.Infof("Still waiting for the cluster-etcd-operator to bootstrap")
 	return false, nil
 }
+
+func WaitForKubeApiServerRollout(ctx context.Context, config *rest.Config) error {
+	operatorClient, err := operatorversionedclient.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("WaitForKubeApiServerRollout: error getting operator client: %#v", err)
+		return err
+	}
+	coreClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("WaitForKubeApiServerRollout: error getting kube client: %#v", err)
+		return err
+	}
+
+	// TODO: figure out if we can timeout after 30 mins by
```

Comment:
Alright, you're structured wrong.  You want to have your sync loop run one and never wait for anything.  Then you wire a time based retrigger of every minute to your control loop. 

The impact of multiple waits is basically impossible to reason about after the fact and doesn't product logs.